ast/functions.go

```diff
@@ -330,7 +330,17 @@ type FuncCallExpr struct {
 
 // Restore implements Node interface.
 func (n *FuncCallExpr) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+	ctx.WritePlainf("%s(", n.FnName.O)
```

Comment:
use `WriteName` to append `FnName.O`

Reply:
`WriteName` will wrapped in quotes, so `CONV('a',16,2)` will become `'CONV'('a',16,2)`. Is the right?

Reply:
it use "`" in default flags  =>`CONV`('a',16,2)

Reply:
I tried to replace `WritePlain` with `WriteName`, but the test failed. The restored statements and source statements are not equal, and their AST is different. `CONV('a',16,2)` and ``` `CONV('a',16,2)` ``` not equal.

Reply:
it's seem like a bug of parser , i will test it and create a issue for it

Reply:
And I think there are some bug:
1. BinaryExpr here that are different from the original statements, mainly `=` and `<>`, which led to the test failure. 
2. In addition, ValueExpr string type in the original statement with encoding, the result of the restore does not contain encoding, which also led to the test failure, such as `_latin1'Müller'` will be restored to `'Müller'`.

Reply:
and reply the bug you report:
1. could you show me a fail case?
2. tidb only support utf8mb4, no matter what charset write in sql, tidb always handle them in utf8mb4, so parser ignore the charset

Reply:
> Use `AVG(test_score)` and `` `AVG`(`test_score`) ``.

got it

Reply:
```
func (ts *testFunctionsSuite) TestXXX(c *C) {
	extractNodeFunc := func(node Node) Node {
		return node.(*SelectStmt).Fields.Fields[0].Expr
	}

	parser := parser.New()
	stmt, err := parser.ParseOneStmt("SELECT CONVERT(_latin1'Müller' USING utf8);", "", "")
	if err != nil {
		c.Fatal(err.Error())
	}
	var sb strings.Builder
	extractNodeFunc(stmt).Restore(NewRestoreCtx(DefaultRestoreFlags, &sb))
	restore := sb.String()
	if restore != "CONVERT(_latin1'Müller' USING utf8)" {
		c.Fatal(sb.String())
	}
}
```

And I get output

```
functions_test.go:117:
    c.Fatal(sb.String())
... Error: CONVERT('Müller' USING utf8)
```

Reply:
> `CONV` parse result is right, but `AVG(test_score)` and `` `AVG`(`test_score`) `` parse result is not equal. But they should be the same. @tiancaiamao

yes, it seems like a bug

Reply:
Well, it looks like I need to wait for this bug fix to move on to the next step.

Reply:
> Well, it looks like I need to wait for this bug fix to move on to the next step.

yes

Reply:
I think I found out the reason for this bug, but I don't know how to fix it.
`parser/ast/misc.go:622`, look back here for a character to confirm whether you need to check the BUILTIN function name, so ``` `AVG`(`test_score`) ``` peek() will get "'" instead of "(".
But I don't know how to fix it, because I didn't find a way to view two characters backwards.

Reply:
I think I found out the reason for this bug, but I don't know how to fix it.
``` `AVG` ``` will be recognized as `-identifier`, so `isTokenIdentifier` method will not be called.